### PR TITLE
First pages translated into Spanish

### PR DESCRIPTION
**Pull request by mberasategi**
## Overview
**TL;DR**
<One to two sentence description of the issue you are encountering or trying to solve.>

As commented on #583, I've begun translating the learning path into Spanish. 

<Link to related issue. Type `closes #RELATEDISSUENUMBER` to establish a link.>

### Questions
<If relevant, write a list of questions that you would like to discuss related to the changes that you have made.>

* Confirm file naming and paths. 
* Is it going to be possible to see something on the interface even if the translation is not complete?

### Next Steps
<If incomplete, create a task list of items that are still being worked on within the Pull Request.>
- [ ] Check if translation works well through the interface
- [x] Keep translating learning path pages.

### Review
<If complete, or ready for :eyes:, use @mentions for quick questions, specific feedback, and progress updates.>

### CODE REVIEW

**Comment by beardofedu** [comment]
@mberasategi so I think we need to make a new `path` for the Spanish content. So currently the Introduction to GitHub course path points to `intro-to-github` and `intro-to-github/page-name`, so I think the Spanish courses should point to `intro-to-github-es` and `intro-to-github/page-name`. After we change the path for the `permalink` and the `next-page` attributes in the different course files, we need to modify the `data/navigation.yml` file by creating a new table of contents for the course which will be used by the course for the side navigation. Overall, this is an incredible first step in the creation of the translated content and your work on this is fantastic 🎉 . Let me know if there is anything you would like me to take on so you can focus on the content translation. 

@hollenberry and @hectorsector for 👀s and 💭 s

**Comment by mberasategi** [comment]
@beardofedu I think you mistyped the paths, you mean `intro-to-github-es` and `intro-to-github-es/page-name`, right? Instead of `intro-to-github/es_ES/page-name` that I created. I will change that and commit in a minute.

I will be more than happy to keep up with the translation, more so if we can start seeing it on the website! I'm doing this because I'm going to use it with students, so doing it as a contribution takes almost the same effort on my part but serves the community and as an example for students of how contributions work, so it's win-win! :wink:

**Comment by hectorsector** [comment]
@mberasategi thanks so much for your contributions, and continuing to work on them ❤️ 

We need to do some work to make sure our site supports localization and all languages. This is work that we have to do on the backend and until it's done you won't be able to preview your pages 😢 . But know that we haven't forgotten about you, and we're working on it! As soon as I have a Pull Request open I'll let you know so you can keep 👀  on it. 

In the meantime, the only workaround I can think of is to serve your site locally, and give each of the spanish pages an `es` permalink. Let me know if you're interested in some help setting that up.

**Comment by mberasategi** [comment]
@hectorsector sure! The first translation is always the hardest :wink:

I am definitely interested in that temporary solution you suggest. Let me know if you want to discuss the details here or find somewhere else to avoid cluttering this thread! Thanks so much!

**Comment by hectorsector** [comment]
I submitted the comment too soon. If you can't spin it up locally, there's just one more step I can walk you through and we can test it by publishing it on your own repository.

**Comment by mberasategi** [comment]
Thanks @hectorsector , however, `NEWBIE ALERT` :mega: I have no idea what I'm doing here :sweat_smile: 

I am on my office computer and I have Windows 7. What do I need to run these? Maybe just an Ubuntu VM?

**Comment by mberasategi** [comment]
OK, done. I installed so many things I can't even remember :sweat_smile: but it's running. Now I need that "one more step" you mentioned because I don't seem to be able to see the pages in Spanish, apart from the first page (http://127.0.0.1:4000/on-demand/intro-to-github-es/). I can't find out how to change those permalinks. 

If that worked, at least I could _show_ them the pages in Spanish. Next step would be to share them somehow so that they can see them for themselves. 

Thanks again!

**Comment by hectorsector** [comment]
@mberasategi apologies for the late response. I was in Hurricane Irma's direct path 😨 All is well now though 😌 

> I have no idea what I'm doing here

Glad you were able to get it to run on your own! 👏 👏 👏 

> I can't find out how to change those permalinks.

Each page should have a section of what's called **YAML front-matter** wrapped in three dashes before and after. These are essentially key-value pairs that we can use to tell Jekyll (the technology running our site) how to behave. One of these keys is called `permalink`, which determines the URL at which to serve each page. You can change the `permalink` and `next-page` keys in the front-matter of the spanish pages. Those pages will then be visible from your web browser at the URLs you specified. I believe you gave me access to edit your fork, so if you can't get it to work I'd be glad to help you do that.

> Next step would be to share them somehow so that they can see them for themselves.

That's the part I need to do on the backend. Localization should be as straightforward as possible, and users shouldn't have to know to go to the `/es/` path, for example, to view a Spanish page. I'm actively looking into how other people have localized content using Jekyll, which is what we need to do as well.

**Comment by DuhhVinci** [comment]
Hello world and everyone who is apart of it  . I'm duhvinci and I new too this world I would be very appreciative it if I can get some guidance  to begin my knowledge nd begin my exploration  of the new world⁉️‼️🕸🤓

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by mberasategi** [comment]
Don't worry @hectorsector , hurricanes are valid enough reasons to take your time to respond. Glad to know all is well now.

That YAML front matter info was key. I have it all working now locally, including localized menu, layout etc. I'm going with translating includes and tell-me-why pages at the moment (not pushing yet though). At least that will allow me to show students something they can read in today's class.

I understand it will take time for this to go live, but is there a way to have it working _online_ somewhere, maybe in my repository, so that students can access from their computers? It will be hard for them to follow otherwise.

Let me know, and thanks!

**Comment by mberasategi** [comment]
Finished translating the intro path, including layout, navigation and includes (activities, tell-me-why's etc). Pushed everything. 

Working well locally, except some activity includes which are not building correctly, apparently. Will keep working on that.

Let me know if you think I could make this available to students somehow.

**Comment by hectorsector** [comment]
Yes! You can make it live in your own repository by going to your Settings tab, down to the GitHub Pages section, and enabling that to serve from your `master` branch. You may have to change the `baseurl` key in the `_config.yml` file so it points to `/mberasategi` instead of `/on-demand`. This should ensure that all references to stylesheets, javascript, etc, point to the proper url.

**Comment by hectorsector** [comment]
Apologies @mberasategi you may not need to change the baseurl after all. GitHub Pages will automatically serve your site at `mberasategi.github.io/REPO-NAME`. Since your repository's name is `training-kit` it will be at `mberasategi.github.io/training-kit` which is what the `baseurl` was already set for. I don't know what I was thinking.

**Comment by mberasategi** [comment]
No @hectorsector , it actually said `/on-demand` :smile: So I changed it to `/training-kit` instead of `/mberasategi` and it's fully working now: https://mberasategi.github.io/training-kit/intro-to-github-es/

Which is EXACTLY what I was hoping for :heart: Off to class now, you may see a bunch of Spanish students going through the intro class in the following couple of hours -- let's hope!!

**Comment by hectorsector** [comment]
It looks fantastic, @mberasategi! I hope class goes ✨

**Comment by hectorsector** [comment]
@mberasategi could you take a look at https://github.com/github/training-kit/pull/590#issuecomment-329312044? I've merged this branch into that one.
